### PR TITLE
check logicalPort in bfdSpec

### PR DESCRIPTION
**Pull request by andriidovhan**

### CODE REVIEW

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/BfdSpec.groovy [hunk 1]
```diff
@@ -71,6 +79,25 @@ class BfdSpec extends HealthCheckSpecification {
             }
         }
 
+        and: "BFD logical ports are created"
+        def srcSwLogPort = isl.srcPort + bfdLogPortDelta
+        def dstSwLogPort = isl.dstPort + bfdLogPortDelta
+        def srcSwitchAddress = northbound.getSwitch(isl.srcSwitch.dpId).address
+        def dstSwitchAddress = northbound.getSwitch(isl.dstSwitch.dpId).address
+        [isl.srcSwitch, isl.dstSwitch].each { sw ->
+            def swAddress = sw.dpId == isl.srcSwitch.dpId ? srcSwitchAddress : dstSwitchAddress
```
Comment (rtretyak) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/BfdSpec.groovy:88]: `def swAddress = northbound.getSwitch(it.dpId).address` and remove L85-86

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/BfdSpec.groovy [hunk 2]
```diff
@@ -71,6 +79,25 @@ class BfdSpec extends HealthCheckSpecification {
...
+            def swPort = sw.dpId == isl.srcSwitch.dpId ? isl.srcPort : isl.dstPort
+            def swLogPort = sw.dpId == isl.srcSwitch.dpId ? srcSwLogPort : dstSwLogPort
```
Comment (rtretyak) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/BfdSpec.groovy:90]: `def swLogPort = swPort + bfdLogPortDelta` and remove L83-84

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/BfdSpec.groovy [hunk 3]
```diff
@@ -248,6 +290,24 @@ class BfdSpec extends HealthCheckSpecification {
...
         }
 
+        and: "BFD logical ports are deleted"
+        when: "Get isl source port"
+        def srcSwLogPort = isl.srcPort + bfdLogPortDelta
+        def srcSwitchAddress = northbound.getSwitch(isl.srcSwitch.dpId).address
+        grpc.getSwitchLogicalPortConfig(srcSwitchAddress, srcSwLogPort)
+
+        then: "Human readable error is returned, because it was deleted while removing BFD session."
+        def e = thrown(HttpServerErrorException)  // issue 3754
```
Comment (rtretyak) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/BfdSpec.groovy:300]: please do `#3754` of full ticket url. Searching project by just a number is painful

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/BfdSpec.groovy (hunk 2 above)
Comment (andriidovhan) [reply src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/BfdSpec.groovy]: fixed

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/BfdSpec.groovy (hunk 3 above)
Comment (andriidovhan) [reply src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/BfdSpec.groovy]: fixed

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/BfdSpec.groovy (hunk 1 above)
Comment (andriidovhan) [reply src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/BfdSpec.groovy:87]: I would like to leave it as is.
These variables are reused in L158 and L165
It allows us to reduce the number of requests to `northbound`.

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/BfdSpec.groovy [hunk 4]
```diff
@@ -31,8 +35,12 @@ Main purpose is to detect ISL failure on switch level, which is times faster tha
 controller-involved discovery mechanism""")
 @Tags([HARDWARE])
 class BfdSpec extends HealthCheckSpecification {
+    @Autowired
+    GrpcService grpc
     @Shared
     BfdProperties defaultBfdProps = new BfdProperties(350, (short)3)
+    @Shared
+    Integer bfdLogPortDelta = 1000
```
[the remark this answers is not in the document]
Comment (andriidovhan) [reply src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/BfdSpec.groovy]: fixed

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/BfdSpec.groovy (hunk 1 above)
Comment (rtretyak) [reply src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/BfdSpec.groovy:87]: ah, didn't notice

**Review by rtretyak — APPROVED** [review]

**Pull request merged**